api/types/github.go

```diff
@@ -218,8 +242,8 @@ func (c *GithubConnectorV3) SetDisplay(display string) {
 
 // MapClaims returns a list of logins based on the provided claims,
 // returns a list of logins and list of kubernetes groups
-func (c *GithubConnectorV3) MapClaims(claims GithubClaims) ([]string, []string, []string) {
-	var logins, kubeGroups, kubeUsers []string
+func (c *GithubConnectorV3) MapClaims(claims GithubClaims) ([]string, []string, []string, []string) {
+	var logins, roles, kubeGroups, kubeUsers []string
```

Comment:
Why don't we just rename `logins` here to `roles`, and append the roles from the new field to the same variable, and keep the function signature the same?

Reply:
Well if we do that, role from the new mapping set would also spill over onto logins or logins would spill onto role mappings from the old set, neither of which seem desirable.

Reply:
Aren't "logins" here basically "roles" though? Just an old terminology we never cleaned up from before the RBAC was open-sourced. You concatenate them in github.go anyway after they're returned from here. No?

Reply:
Sort of but also not really. Before RBAC for OSS, this field assigned allowed *nix logins only. When we introduced RBAC for OSS we made the decision to make the GH connector work with it, do to that we added a hack that assigns the login field to **both** allowed *nix loginx *and* roles. So we don't really concatenate them. This behaviour is fundamentally different since mappings from the new set should only translate to roles without getting duplicated on the *nix login list.

Reply:
Sounds good.